aspnetcore/fundamentals/openapi/aspnetcore-openapi.md

```diff
@@ -344,6 +344,105 @@ app.MapGet("/attributes",
   () => "Hello world!");
 ```
 
+## Including OpenAPI metadata for data types
+
+C# classes or records used in request or response bodies are represented as schemas
+in the generated OpenAPI document.
+By default, only public properties are represented in the schema, but there are
+<xref:System.Text.Json.JsonSerializerOptions> to also create schema properties for fields.
+
+When the <xref:System.Text.Json.JsonSerializerOptions.PropertyNamingPolicy> is set to camel-case (this is the default
+in ASP.NET web applications), property names in a schema are the camel-case form
+of the class or record property name.
+The <xref:System.Text.Json.Serialization.JsonPropertyNameAttribute> can be used on an individual property to specify the name
+of the property in the schema.
+
+## type and format
+
+The JSON Schema library maps standard C# types to OpenAPI `type` and `format` as follows:
+
+| C# Type        | OpenAPI `type` | OpenAPI `format` |
+| -------------- | -------------- | ---------------- |
+| int            | integer        | int32            |
+| long           | integer        | int64            |
+| short          | integer        | int16            |
+| byte           | integer        | uint8            |
+| float          | number         | float            |
+| double         | number         | double           |
+| decimal        | number         | double           |
+| bool           | boolean        |                  |
+| string         | string         |                  |
+| char           | string         | char             |
+| byte[]         | string         | byte             |
+| DateTimeOffset | string         | date-time        |
+| DateOnly       | string         | date             |
+| TimeOnly       | string         | time             |
+| Uri            | string         | uri              |
+| Guid           | string         | uuid             |
+| object         | _omitted_      |                  |
+| dynamic        | _omitted_      |                  |
+
+Note that object and dynamic types have _no_ type defined in the OpenAPI because these can contain data of any type, including primitive types like int or string.
+
+The `type` and `format` can also be set with a [Schema Transformer](#use-schema-transformers). For example, you may want the `format` of decimal types to be `decimal` instead of `double`.
```

Comment:
Do we intend to embed the sample code from above in here?

Reply:
It is included in the "Use schema transformers" section.